js/visualizer.js

```diff
@@ -107,55 +143,91 @@ const Visualizer = (fps) =>
         const floorInput  = document.createElement('input');
         floorInput.type = 'checkbox';
         floorInput.checked = true;
         floorInput.dataset.window = windowElem.id;
         floorToggle.classList.add('floor-toggle');
-        floorToggle.appendChild(document.createTextNode('display floor'));
+        floorToggle.appendChild(document.createTextNode('Display Floor'));
         floorToggle.appendChild(floorInput);
         windowElem.appendChild(floorToggle);
 
+		// Create checkbox for toggling camera lock.
+        const cameraToggle = document.createElement('p');
+        const cameraInput  = document.createElement('input');
+        cameraInput.type = 'checkbox';
+        cameraInput.checked = true;
+        cameraInput.dataset.window = windowElem.id;
+        cameraToggle.classList.add('camera-toggle');
+        cameraToggle.appendChild(document.createTextNode('Cam Control'));
+        cameraToggle.appendChild(cameraInput);
+        windowElem.appendChild(cameraToggle);
+
+		// Set default camera position.
         camera.position.set(600, 600, 1000);
         camera.lookAt(new THREE.Vector3(0,0,0));
 
-        // Directional light to enhance shadow.
-        let defaultLight = new THREE.DirectionalLight(0xffffff);
-        defaultLight.position.set(0, 1000, 0);
-        defaultLight.lookAt(new THREE.Vector3(0, 0, 0));
-        scene.add(defaultLight);
+		// Spotlight.
+		spotLight = new THREE.SpotLight();
+		spotLight.position.set(0,1000,0);
+		spotLight.angle = .4;
+		spotLight.penumbra = 1;
+		spotLight.castShadow = true;
+		spotLight.distance = 2000;
+		scene.add(spotLight);
+
+		// Directional light to enhance shadow.
+        let defaultLight1 = new THREE.DirectionalLight(0xffffff, 0.4);
+		defaultLight1.castShadow = true;
+        defaultLight1.position.set(500, 1000, 500);
+        scene.add(defaultLight1);
+
+		let defaultLight2 = new THREE.DirectionalLight(0xffffff, 0.4);
+		defaultLight2.castShadow = true;
+        defaultLight2.position.set(-500, -1000, -500);
+        scene.add(defaultLight2);
 
         // Ambient light to make sure contrast isn't drastic.
-        let ambientLight = new THREE.AmbientLight(0xffffff, 0.5);
+        let ambientLight = new THREE.AmbientLight(0xffffff, 0.3);
         scene.add(ambientLight);
 
-    		// Scene background.
-    		renderer.gammaInput = true;
-    		renderer.gammaOutput = true;
-    		renderer.setClearColor(0x001a0d, 1.0);
-    		renderer.shadowMap.enabled = true;
-
-    		// Grid floor and fog to add perspective for model movement.
-    		scene.fog = new THREE.Fog(0x001a0d, 1500, 4500);
-    		grid = new THREE.GridHelper(10000, 100, 0x001a0d, 0x006633);
-    		scene.add(grid);
+    	// Scene background.
+    	renderer.gammaInput = true;
+    	renderer.gammaOutput = true;
+    	renderer.setClearColor(0x001a0d, 1.0);
+    	renderer.shadowMap.enabled = true;
+		/*
+		let skyGeo = new THREE.SphereGeometry(3000, 60, 40);
```

Comment:
We shouldn't commit commented out code. Save this for later until we know we need it.